src/components/like/LikeButton.js

```diff
@@ -15,19 +15,32 @@ import { getLikePost } from '../../redux/actions/dataActions';
  * @param {object} props
  * @param {string} props.postId
  * @param {any} props.getLikePost
+ * @param {object} props.UI
  */
 export const LikeButton = ({ postId, getLikePost, UI = {} }) => {
+  const makeCall = useRef(false);
   const likePost = () => {
+    if (!!makeCall.current) return;
+    makeCall.current = true;
     getLikePost(postId);
+    setTimeout(() => {
+      makeCall.current = false;
+    }, 3000);
```

Comment:
this is a clever solution. On thinking about it though, shouldn't this logic be something more like:
```
if(!inProgress){
  inProgress:true
  axios.put('post/',{whatever data blabla})
  .then((data)=>{
    inProgress:false
    whatever logic with that data
})
}
```